HeatStroke/GOComponents/GameObjectManager.cpp

```diff
@@ -215,6 +215,23 @@ void GameObjectManager::DestroyGameObject(GameObject* p_pGameObject)
 	m_vToDelete.insert(p_pGameObject);
 }
 
+void GameObjectManager::ForceInstantDestroyGameObject(GameObject* p_pGameObject)
```

Comment:
I'm quite wary of this method existing and the way you use it, but you've also made a TODO comment that you want to find a better way, so I'll leave it at that.